cmd/dp-observation-importer/main.go

```diff
@@ -82,69 +159,137 @@ func main() {
 	observationStore := observation.NewStore(dimensionIDCache, graphDB, errorReporter)
 
 	// write import results to kafka topic.
-	resultWriter := observation.NewResultWriter(kafkaResultProducer)
+	resultWriter := observation.NewResultWriter(observationsImportedProducer)
 
 	// handle a batch of events.
 	batchHandler := event.NewBatchHandler(observationMapper, observationStore, resultWriter, errorReporter)
 
 	eventConsumer := event.NewConsumer()
 
 	// Start listening for event messages.
-	eventConsumer.Consume(kafkaConsumer, cfg.BatchSize, batchHandler, cfg.BatchWaitTime, errorChannel)
+	eventConsumer.Consume(syncConsumerGroup, cfg.BatchSize, batchHandler, cfg.BatchWaitTime, errorChannel)
 
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+	syncConsumerGroup.Channels().LogErrors(ctx, "Consumer error")
+	observationsImportedProducer.Channels().LogErrors(ctx, "Observations Imported Producer error")
+	observationsImportedErrProducer.Channels().LogErrors(ctx, "Observation Imported Error Producer error")
+
+	go func() {
+		select {
```

Comment:
If an error occurs in the HTTP server it logs a message but does not appear to trigger anything else to happen? Should this case be integrated into the select below? That was a http error will cause the same shutdown process as the os signal?

Reply:
Im not sure, I thought the server might be recoverable.

The platform will shutdown the application if it deems the app is not fit for purpose, by calling the health check endpoint; if this is not accessible due to a continuous issue with the server then it will close the application and restart it. Also because the application only provides http requests for health endpoint I thought it makes sense for it not to shut down the application

Reply:
After delving into go-ns server and then http server I think you are right it will not recover, so we should start gracefully shutting down.